src/groundlight/client.py

```diff
@@ -270,6 +332,37 @@ def wait_for_confident_result(
             image_query = self._fixup_image_query(image_query)
         return image_query
 
+    def _wait_for_ml_result(self, image_query: Union[ImageQuery, str], timeout_sec: float = 30.0) -> ImageQuery:
```

Comment:
Can we merge this with “wait_for_confident”?  I imagine there’s a lot of duplicated logic in terms of retry and back off.

Reply:
I merged them in an earlier draft but actually opted for duplicity in favor of better logging. I'm going back to the previous version with them merged but would love any tips on how to provide the same level of specificity while logging if the combined logic is indifferent to being called from wait_for_ml or from wait_for_confident